crates/view/src/storage.rs

```diff
@@ -1493,4 +1500,54 @@ impl Storage {
         })
         .await?
     }
+
+    pub async fn amount_by_sender(
+        &self,
+        start_height: Option<u64>,
+        end_height: Option<u64>,
+    ) -> anyhow::Result<BTreeMap<(Address, String), u128>> {
+        let starting_block = start_height.unwrap_or(0) as i64;
+        let ending_block = end_height.unwrap_or(self.last_sync_height().await?.unwrap_or(0)) as i64;
+
+        let pool = self.pool.clone();
+
+        let query = "SELECT return_address, denom, amount
+                    FROM tx
+                    JOIN spendable_notes ON spendable_notes.source = tx.tx_hash
+                    JOIN notes ON spendable_notes.note_commitment = notes.note_commitment
+                    JOIN assets ON assets.asset_id = notes.asset_id
+                    WHERE block_height BETWEEN ?1 AND ?2
+                    GROUP BY return_address";
```

Comment:
What does GROUP BY do for Blobs?

Reply:
Without grouping by the address, you won't filter out asset amounts for the other addresses, because the JOIN does not otherwise take this into account/is not unique to the address. I initially thought the same but testing revealed the need to specify a specific grouping to return the appropriate values per address.

Reply:
Have we tested the case where you send the exact same amount twice? I think the GROUP BY might clobber that

Reply:
Like my understanding of how GROUP BY works without aggregation functions is that it's like doing SELECT DISTINCT, which would have this undesired behavior when you see a repeat of (return_addr, denom, amount)